CppCoreGuidelines.md

```diff
@@ -9897,9 +9897,13 @@ Avoid wrong results.
 
 ##### Example
 
-    unsigned x = 100;
-    unsigned y = 102;
-    cout << abs(x-y) << '\n'; // wrong result
+    while(file.read((char*)&obj, sizeof obj))
+        if (obj.val == bad_value)  {
```

Comment:
cast needed if we're sticking to realistic example (scanning a binary file and making the mistake when calculating a negative offset). Changed to `reinterpret_cast`..